server/src/main/java/org/elasticsearch/rest/action/cat/RestNodesAction.java

```diff
@@ -86,6 +92,9 @@ protected void documentation(StringBuilder sb) {
     public RestChannelConsumer doCatRequest(final RestRequest request, final NodeClient client) {
         final ClusterStateRequest clusterStateRequest = new ClusterStateRequest();
         clusterStateRequest.clear().nodes(true);
+        if (request.hasParam("local")) {
+            deprecationLogger.deprecatedAndMaybeLog("cat_nodes_local_deprecation", LOCAL_DEPRECATED_MESSAGE);
```

Comment:
`deprecatedAndMaybeLog` performs some deduplication but I think we should always emit this warning, i.e. call `deprecated` directly.